pkg/fsnotify/fsnotify.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package fsnotify
+
+import (
+	"context"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/shirou/gopsutil/v3/disk"
+	"go.uber.org/zap"
+
+	"github.com/iotexproject/iotex-core/pkg/log"
+)
+
+// Watch creates dir watcher to hanlde events from os in loop
+func Watch(ctx context.Context, dirs ...string) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.L().Fatal("Failed to create fsnotify watcher.", zap.Error(err))
+	}
+	defer watcher.Close()
+
+	for _, dir := range dirs {
+		if err = watcher.Add(dir); err != nil {
+			log.L().Fatal("Failed to watch dir: "+dir, zap.Error(err))
+		}
+	}
+	log.S().Infof("Watch dirs: %+v", dirs)
+
+	watchLoop(ctx, watcher)
+}
+
+// watchLoop receives events and errors from os
+func watchLoop(ctx context.Context, watcher *fsnotify.Watcher) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case ev, ok := <-watcher.Events:
+			if !ok {
+				log.L().Error("Failed to get watcher events.")
+				continue
+			}
+			// check disk space per write operation
+			if ev.Has(fsnotify.Write) {
```

Comment:
fsnotify.Create needed too

Reply:
It's still left enough space to run program  even if `Avail` is `0` after run `df -h` .